### PR TITLE
docs: Add banner to v1 and v2 docs indicating that v1 is archived

### PR DESCRIPTION
**Pull request by maggieneterval**
Fixes: https://github.com/GoogleContainerTools/skaffold/issues/7910
Related: https://github.com/GoogleContainerTools/skaffold/pull/7917

**Description**
* Add a banner to all v1 documentation pages indicating that v1 is archived, using [Docsy's built-in functionality of displaying banners on archived docs sites](https://www.docsy.dev/docs/adding-content/versioning/#displaying-a-banner-on-archived-doc-sites). (Note: this is similar to [archived Kubernetes documentation](https://v1-21.docs.kubernetes.io/docs/tutorials/configuration/)).
* Add a banner to the main v2 documentation page indicating that v2 has been released, and providing a link back to the archived v1 documentation.

**User facing changes**

V1 screenshot:

mh4NS2E4BkpSKTV

V2 screenshot:

8JZzHFfgRr79bKX

### CODE REVIEW

**Comment by maggieneterval** [comment]
@aaron-prindle Wording/placement suggestions welcome! LMK if you would like to provide a release date on the v2 banner.

**Review by tejal29 — APPROVED** [review]

**Comment by tejal29** [comment]
Look great! Will wait for @aaron-prindle to LGTM before merge.

**Pull request merged**
